### PR TITLE
Add support for passing environment variables to terraform tasks

### PR DESCRIPTION
**Pull request by JayDoubleu**
### Change description ###

Add support for passing environment variables to terraform tasks

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by JayDoubleu** [comment]
@timja  sure, README.md updated now.
 
See below sample run without those env vars set. ( proves the template still works without them however terraform breaks due to missing TF_VAR)

https://github.com/hmcts/dynatrace-availability-dashboards/pull/10

File: README.md [hunk 1]
```diff
@@ -129,3 +129,23 @@ tfVarsFile: NULL
...
+          - template: steps/terraform.yaml@cnp-azuredevops-libraries
+            parameters:
+              overrideAction: ${{ parameters.overrideAction }}
+              environment: ${{ deployment.environment }}
+              component: ${{ deployment.component }}
+              serviceConnection: ${{ deployment.service_connection }}
+              terraformInitSubscription: ${{ variables.terraformInitSubscription }}
+              product: ${{ variables.product }}
+              tf_env_vars:
+                TF_VAR_FOO: $(bar)
```
Comment (timja) [inline README.md:148]: should be lowercase I think so that you get a variable that matches the naming standard
```suggestion
                TF_VAR_foo: $(bar)
```

File: README.md [hunk 2]
```diff
@@ -129,3 +129,23 @@ tfVarsFile: NULL
...
+```
+In terraform you can then reference this variable as `var.FOO`
```
Comment (timja) [inline README.md:150]:
```suggestion
In terraform you can then reference this variable as `var.foo`
```

File: README.md [hunk 3]
```diff
@@ -129,3 +129,23 @@ tfVarsFile: NULL
...
 
 > see [example in aks-cft-deploy repo](https://github.com/hmcts/aks-cft-deploy/blob/main/azure-pipelines.yml)
+
+### Passing environment variables to terraform template:
+
+You can pass environment variables directly to terraform tasks (plan, apply, destroy)
+Which then can be used as variable within terraform code as shown in below example:
+
+```yaml
+          - template: steps/terraform.yaml@cnp-azuredevops-libraries
```
Comment (timja) [inline README.md:139]: can you indent this all the way back to the left

File: README.md [hunk 4]
```diff
@@ -129,3 +129,23 @@ tfVarsFile: NULL
...
+            parameters:
+              overrideAction: ${{ parameters.overrideAction }}
+              environment: ${{ deployment.environment }}
+              component: ${{ deployment.component }}
+              serviceConnection: ${{ deployment.service_connection }}
+              terraformInitSubscription: ${{ variables.terraformInitSubscription }}
+              product: ${{ variables.product }}
+              tf_env_vars:
```
Comment (timja) [inline README.md:147]: please be consistent with other variable naming in this project
```suggestion
              terraformEnvironmentVariables:
```

File: README.md [hunk 5]
```diff
@@ -4,31 +4,31 @@ Reusable pipeline components for CNP Azure DevOps pipelines
...
 
 ### Reusing templates:
 1. Create the [required folder structure](#required-terraform-folder-structure) in your repository
-2. Add the cnp-azure-devops-libraries repository resource as below  
-   
+2. Add the cnp-azure-devops-libraries repository resource as below
+
    ```yaml
-   
-   resources:
```
Comment (timja) [inline README.md:21]: please make unrelated formatting changes in a separate pull request or disable your IDE trimming unrelated changes

**Review by timja — APPROVED** [review]
LGTM apart from unrelated whitespace, but will approve anyway just keep that in mind for future please

File: README.md (hunk 5 above)
Comment (JayDoubleu) [reply README.md:21]: 👍
